src/clients/firewall.rb

```diff
@@ -57,11 +57,34 @@ def main
         SuSEFirewallCMDLine.Run 
         # GUI or TextUI
       else
-        # installation has other sequence
-        if Mode.installation
-          @ret = FirewallInstallationSequence()
+        # If FirewallD then use it's UI
+        if SuSEFirewall.is_a?(Yast::SuSEFirewalldClass)
```

Comment:
ah, now I see that here is also used suseFirewalldClass, so also here should be require at top